### PR TITLE
FEAT: ECS implementation and benchmark

### PR DESCRIPTION
**Pull request by mathieurobert1**
I made the Benchmark for ECS/OOP at this link : [SlimWiki](https://slimwiki.com/6fvu0f5pu/getting-started-nqdxve9qg-/ecs-design-pattern-benchmark-rox0gh-gl-q8er45r7ckq2)

I implemented the ECS. It's a static library build in the folder "Ecs".

With ECS you can add component and entities, and manage them with systems.

>[!IMPORTANT]
> There no tests for ECS while Cmake is not configure to made tests.
> issue : #19

### CODE REVIEW

File: Ecs/src/Registry.hpp
Comment (Thomaltarix) [inline Ecs/src/Registry.hpp:20]: I think you can add comments to these 2 lines

File: Ecs/src/SparseArray.hpp
Comment (Thomaltarix) [inline Ecs/src/SparseArray.hpp:24]: I think you can also comment this part of code

**Review by AxelF44 — APPROVED** [review]

File: Ecs/src/SparseArray.hpp
Comment (Thomaltarix) [inline Ecs/src/SparseArray.hpp:42]: I think it would be a great idea to discuss the usage of "this" to have the same logic in all the project. For me, it's useless to use the "this" specifier for private variables, since their type is already defined by the underscore before the variable name. I'll discuss with Axel about this today 👍

Comment (Thomaltarix) [reply Ecs/src/SparseArray.hpp:42]: I'll not say this for the all this specifiers, but you know what i mean

**Review by Thomaltarix — APPROVED** [review]

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
You missed some comments at some places

File: Ecs/src/SparseArray.hpp
Comment (mathieurobert1) [reply Ecs/src/SparseArray.hpp]: Finally, I only use this-> for method call and not private variables since their type is already defined by the underscore.

**Review by AxelF44 — APPROVED** [review]

File: Ecs/src/SparseArray.hpp
Comment (Thomaltarix) [reply Ecs/src/SparseArray.hpp]: Seen with @AxelF44, I'll update the Contributing soon

File: Ecs/src/SparseArray.hpp
Comment (Thomaltarix) [inline Ecs/src/SparseArray.hpp:300]: Since in other methods you put brackets for one line execution if, I think, as the "this" specifier, we should specify in the contributing if we are keeping the brackets or not for single line if conditions

**Review by Thomaltarix — COMMENTED** [review]
I left a little comment, i'll let you check it and approve just after

**Review by Thomaltarix — APPROVED** [review]

**Pull request merged**
